File: tests/integration-tests/src/main/java/com/jaynewstrom/json/sample/custom/LenientBooleanDeserializer.java
package com.jaynewstrom.json.sample.custom;

import com.fasterxml.jackson.core.JsonParser;
import com.jaynewstrom.json.runtime.JsonDeserializer;
import com.jaynewstrom.json.runtime.JsonDeserializerFactory;

import java.io.IOException;

final class LenientBooleanDeserializer implements JsonDeserializer<Boolean> {
    @Override public Boolean deserialize(JsonParser jp, JsonDeserializerFactory deserializerFactory) throws IOException {
        return jp.getValueAsBoolean();
    }
}

File: tests/integration-tests/src/main/java/com/jaynewstrom/json/sample/custom/LenientBooleanDeserializer.kt
package com.jaynewstrom.json.sample.custom

import com.fasterxml.jackson.core.JsonParser
import com.jaynewstrom.json.runtime.JsonDeserializer
import com.jaynewstrom.json.runtime.JsonDeserializerFactory

import java.io.IOException

internal object LenientBooleanDeserializer : JsonDeserializer<Boolean> {
    @Throws(IOException::class)
    override fun deserialize(jp: JsonParser, deserializerFactory: JsonDeserializerFactory): Boolean = jp.valueAsBoolean
}
